lib/actions/app.coffee

```diff
@@ -134,9 +141,24 @@ exports.remove =
 	options: [ commandOptions.yes ]
 	permission: 'user'
 	action: (params, options, done) ->
-		visuals.patterns.remove 'application', options.yes, (callback) ->
-			resin.models.application.remove(params.name).nodeify(callback)
+		async.waterfall([
+
+			(callback) ->
+				if options.yes
+					return callback(null, true)
+				else
+					form.ask
+						message: 'Are you sure you want to delete the application?'
+						name: 'confirm'
+						type: 'confirm'
+						default: false
+					.nodeify(callback)
+
+			(confirmed, callback) ->
+				return callback() if not confirmed
+				resin.models.application.remove(params.name).nodeify(callback)
 		, done
```

Comment:
This `, done` here makes no sense. It should be the second argument to the `async.waterfall()` call instead.